### PR TITLE
feat(modal): add modal component

### PR DESCRIPTION
**Pull request by rossmoody**
## Description

This PR adds the Modal component to the website. 

## Checklist

- [ ] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [ ] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [ ] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [ ] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] :zap: audited via [web.dev](https://web.dev/measure/) for performance and SEO
- [ ] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by jzempel** [comment]
Something borked with the deployment. Re-running now...

**Review by hzhu — COMMENTED** [review]
It looks like there's an issue with the focusvisible behavior when the modal is opened. The focus outline isn't removed on blur.

Edit: after some investigation, it seems to be a [regression](https://zendeskgarden.github.io/react-components/modals/) unrelated to this PR. The root cause of the problem is TBD.

Edit 2: Looks like could be an issue with [react-merge-refs](https://github.com/zendeskgarden/react-components/pull/876/files#diff-b1dcdbfc5df6b70c458a6ffc64e4c2a6R154). A ref callback is being passed to `useFocusVisible` which means the focusvisible `scope` is undefined. I think it should take a ref object (rather than ref callback).

File: src/examples/modal/ModalDanger.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { Modal, Header, Body, Footer, FooterItem, Close } from '@zendeskgarden/react-modals';
+import { Row, Col } from '@zendeskgarden/react-grid';
+
+const Example = () => {
+  const [visible, setVisible] = useState(false);
+
+  return (
+    <Row>
+      <Col textAlign="center">
+        <Button onClick={() => setVisible(true)}>Open modal</Button>
```
Comment (jzempel) [inline src/examples/modal/ModalDanger.tsx:19]: Should this button be `isDanger` to differentiate from the other examples?

**Review by jzempel — APPROVED** [review]
:tada:

**Review by hzhu — APPROVED** [review]

**Comment by rossmoody** [comment]
@vsrixyz @allisonacs This one was in limbo for a while due to build problems but it's ready for a look now.

File: src/examples/modal/ModalDanger.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    <Row>
+      <Col textAlign="center">
+        <Button onClick={() => setVisible(true)} isDanger>
+          Open danger modal
+        </Button>
+        {visible && (
+          <Modal onClose={() => setVisible(false)}>
+            <Header isDanger>Remove plant food from your garden</Header>
+            <Body>
+              Plant food is great for the health of a garden. Are you sure you want to remove it?
```
Comment (allisonacs) [inline src/examples/modal/ModalDanger.tsx:26]: We don't ask users if they are sure in Zendesk. Per our tone guidelines, we tell people what the effects of a destructive action are. Perhaps something more like

> Plant food helps gardens grow. Removing plant food negatively affects the plant health.

@vsrixyz what do you think?

**Review by allisonacs — CHANGES_REQUESTED** [review]
Couple of content issues here, but implementation looks good.

File: src/examples/modal/ModalDanger.tsx (hunk 2 above)
Comment (rossmoody) [reply src/examples/modal/ModalDanger.tsx:26]: Made these changes for now to keep this moving.

**Review by allisonacs — APPROVED** [review]

**Pull request merged**